src/DoctrineMongoODMModule/Paginator/Adapter/DoctrinePaginator.php

```diff
@@ -49,6 +50,11 @@ public function __construct(Cursor $cursor)
      */
     public function count()
     {
+        // Avoid using EagerCursor::count as this stores a collection without limits to memory
+        if ($this->cursor->getBaseCursor() instanceof EagerCursor) {
```

Comment:
ODM\EagerCursor is deprecated and will be removed - the eager functionality is contained within MongoDB\EagerCursor.